GVFS/GVFS/CommandLine/PrefetchVerb.cs

```diff
@@ -226,32 +264,75 @@ private void PrefetchCommits(ITracer tracer, GVFSEnlistment enlistment, GitObjec
             }
         }
 
-        private void PrefetchBlobs(ITracer tracer, GVFSEnlistment enlistment, GitObjectsHttpRequestor blobRequestor, CacheServerInfo cacheServer)
+        private void LoadBlobPrefetchArgs(
+            ITracer tracer,
+            GVFSEnlistment enlistment,
+            out string headCommitId,
+            out List<string> filesList,
+            out List<string> foldersList,
+            out FileBasedDictionary<string, string> lastPrefetchArgs)
         {
-            BlobPrefetcher blobPrefetcher = new BlobPrefetcher(
-                tracer,
-                enlistment,
-                blobRequestor,
-                ChunkSize,
-                SearchThreadCount,
-                DownloadThreadCount,
-                IndexThreadCount);
-
             string error;
-            if (!BlobPrefetcher.TryLoadFolderList(enlistment, this.Folders, this.FoldersListFile, blobPrefetcher.FolderList, out error))
+            
+            if (!FileBasedDictionary<string, string>.TryCreate(
+                    tracer,
+                    Path.Combine(enlistment.DotGVFSRoot, "LastBlobPrefetch.dat"),

```

Comment:
Should this go into the .gvfs/databases folder or the .gvfs root?

Reply:
This file has the same format as the other files in the `databases` folder, but those are required files for correct functionality, and this is just a convenient way to store the last prefetch args, not required for anything to work right. So no, I don't think it should go in the same folder.

We don't have any other files that look like this one, so I didn't see a logical place to put it other than the `.gvfs` root. The nice thing is that if we ever decide to move it, it's a non-breaking change. Just delete the old one and let the next prefetch recreate it wherever needed.